lib/magic-byte.js

```diff
@@ -45,24 +44,20 @@ magicByte.toMessageBuffer = function (val, type, schemaId, optLength) {
  *   @param {number} schemaId The schema id.
  *   @param {Object} value The decoded avro value.
  */
-magicByte.fromMessageBuffer = function (type, encodedMessage, sr) {
-  if (encodedMessage[0] !== MAGIC_BYTE) {
-    throw new TypeError('Message not serialized with magic byte');
+magicByte.fromMessageBuffer = function (encodedMessage, sr) {
+  if (!encodedMessage || encodedMessage[0] !== MAGIC_BYTE) {
+    return null;
   }
 
-  var schemaId = encodedMessage.readInt32BE(1);
+  const schemaId = encodedMessage.readInt32BE(1);
+  const schemaType = sr.schemaTypeById[('schema-' + schemaId)];
 
-  var schemaKey = 'schema-' + schemaId;
-  var decoded;
-  if (!sr.schemaTypeById[schemaKey]) {
-    // use default type
-    decoded = type.decode(encodedMessage, 5);
-  } else {
-    decoded = sr.schemaTypeById[schemaKey].decode(encodedMessage, 5);
+  if (!schemaType) {
+    return null;
```

Comment:
this shouldnt be `return type.decode(encodedMessage, 5);` to keep the previous behavior?

Reply:
I have changed the behaviour here. It was unnecessary since all we need is just the magic byte and `sr.schemaTypeById` map.